sim/robots/flood_fill.py

```diff
@@ -634,6 +732,7 @@ def flood_fill_thorough_explorer(  # pylint: disable=too-many-branches,too-many-
         assert next(return_bot, None) is Action.READY
         pos = yield Action.READY
         while True:
+            maze.extra_info[pos[:-1]].reset_color_if()
```

Comment:
I think you can just reset without the `if` no?
```suggestion
            maze.extra_info[pos[:-1]].color = None
```